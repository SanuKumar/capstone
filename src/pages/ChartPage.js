import React from 'react'
import { Bar } from "react-chartjs-2";
import Chart from 'chart.js/auto';
import { Container } from 'react-bootstrap';

const ChartPage = ({ products }) => {
	const data = {
		labels: products.map((product) => product.title),
		datasets: [
			{
				label: "Price",
				data: products.map((product) => product.quantity),
				fill: true,
				backgroundColor: "rgb(255, 99, 132)"
			},
			{
				label: "Rating",
				data: products.map((product) => product.rating),
				fill: false,
				backgroundColor: "rgb(54, 162, 235)"
			}
		]
	};
	return (
		<>
			<Container>
				<div><h2>Top Rated Product</h2></div>
				<Bar data={data} />
			</Container>
		</>
	)
}

export default ChartPage