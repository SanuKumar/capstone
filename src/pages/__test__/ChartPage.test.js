import ChartPage from '../ChartPage';
import { cleanup, render } from "@testing-library/react"

afterEach(cleanup)

test('renders Chart correctly when there are products', () => {
  const products = [{
    "id": 2,
    "name": "iPhone X",
    "description": "SIM-Free, Model A19211 6.5-inch Super Retina HD display with OLED technology A12 Bionic chip with ...",
    "price": 899,
    "rating": 2,
    "quantity": 34,
    "manufacture": "Apple",
    "category": "smartphones",
    "thumbnail": "https://dummyjson.com/image/i/products/2/thumbnail.jpg",
    "images": [
      "https://dummyjson.com/image/i/products/2/1.jpg",
      "https://dummyjson.com/image/i/products/2/2.jpg",
      "https://dummyjson.com/image/i/products/2/3.jpg",
      "https://dummyjson.com/image/i/products/2/thumbnail.jpg"
    ]
  },
  {
    "id": 3,
    "name": "Samsung Universe 9",
    "description": "Samsung's new variant which goes beyond Galaxy to the Universe",
    "price": 1249,
    "rating": 3,
    "quantity": 36,
    "manufacture": "Samsung",
    "category": "smartphones",
    "thumbnail": "https://dummyjson.com/image/i/products/3/thumbnail.jpg",
    "images": [
      "https://dummyjson.com/image/i/products/3/1.jpg"
    ]
  }]
  const { asFragment } = render(<ChartPage products={products} />)
  expect(asFragment()).toMatchSnapshot();
});

