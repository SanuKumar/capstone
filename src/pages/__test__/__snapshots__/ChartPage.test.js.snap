// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Chart correctly when there are products 1`] = `
<DocumentFragment>
  <div
    style="text-align: center; margin-top: -4rem;"
  >
    <h2>
      Top Rated Product Above 3 Stars
    </h2>
  </div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-10 col-md-8 col-sm-12 col-12"
      >
        <canvas
          height="150"
          role="img"
          width="300"
        />
      </div>
      <div
        class="col"
      />
    </div>
  </div>
</DocumentFragment>
`;
